### PR TITLE
Added EventSetup module timing to Timing Service

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Accumulate total time spent in all EventSetup modules. The time is over all threads so can be greater than loop time.

#### PR validation:

Code compiles, framework unit tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Services/plugins/Timing.cc [hunk 1]
```diff
@@ -310,6 +316,42 @@ namespace edm {
...
+          // so should check starting over again
+        } while (not foundSlot);
+      });
+      iRegistry.postESModuleSignal_.connect([this](auto const& key, auto const& context) {
+        auto stopTime = getTime();
+        for (size_t i = 0; i < eventSetupModuleStartTimes_.size(); ++i) {
+          auto const& info = eventSetupModuleCallInfo_[i];
+          if (info.first == uintptr_t(context.componentDescription()) and info.second == key) {
+            auto startTime = eventSetupModuleStartTimes_[i]->load();
+            eventSetupModuleStartTimes_[i]->store(0.);
```
Comment (makortel) [inline FWCore/Services/plugins/Timing.cc:344]: Would
```suggestion
            auto startTime = eventSetupModuleStartTimes_[i]->exchange(0.);
```
make sense?

File: FWCore/Services/plugins/Timing.cc [hunk 2]
```diff
@@ -440,6 +488,7 @@ namespace edm {
                                  << " - Total job:   " << total_job_time << "\n"
                                  << " - EventSetup Lock: " << accumulatedTimeForLock_ << "\n"
                                  << " - EventSetup Get:  " << accumulatedTimeForGet_ << "\n"
+                                 << " - Total ES:    " << accumulatedEventSetupModuleTimings_.load() << "\n"
```
Comment (makortel) [inline FWCore/Services/plugins/Timing.cc:491]: How about
```suggestion
                                 << " - Total EventSetup:    " << accumulatedEventSetupModuleTimings_.load() << "\n"
```
?

I expect this line to create a conflict with https://github.com/cms-sw/cmssw/pull/41023 (trivially resolvable, but still)

File: FWCore/Services/plugins/Timing.cc [hunk 3]
```diff
@@ -310,6 +316,42 @@ namespace edm {
         iRegistry.watchPostSourceConstruction(this, &Timing::postModule);
       }
 
+      iRegistry.preESModuleSignal_.connect([this](auto const& key, auto const& context) {
```
Comment (makortel) [inline FWCore/Services/plugins/Timing.cc:319]: How about
```suggestion
      iRegistry.preESModuleSignal_.connect([this](auto const& recordKey, auto const& context) {
```
to make the purpose of the first argument more clear without having to look the `ActivityRegistry::preESModuleSignal_`?
(I was also thinking to suggest adding explicit types, but then concluded their use below is simple enough to be understood with `auto` types)

**Comment by makortel** [comment]
As anticipated, this PR has now conflicts.

**Comment by Dr15Jones** [comment]
Superseded by #41052

**Pull request closed without merging**